src/main.rs

```diff
@@ -104,17 +104,20 @@ async fn post_middleware_handler(mut resp: Response<Body>) -> Result<Response<Bo
 async fn shutdown_signal() {
     let mut interrupt = signal(SignalKind::interrupt()).unwrap();
     let mut terminate = signal(SignalKind::terminate()).unwrap();
-    tokio::select! {
-        _ = interrupt.recv() => {
-            info!("received interrupt signal");
-        },
-        _ = terminate.recv() => {
-            info!("received terminate signal");
-        },
-    };
-    let task_manager = TaskManager::global();
-    task_manager.shutdown().await;
-    info!("shutdown complete");
+    loop {
+        tokio::select! {
+            _ = interrupt.recv() => {
+                info!("received interrupt signal and ignored at controller side");
+            },
```

Comment:
Why `SIGINT` doesn't need to be shutdown gracefully?

Reply:
It's not that `SIGINT ` does not need graceful shutdown; it's that only `SIGTERM` should shutdown the controller, because it should be controlled by the py side on the shutdown sequence right? If `SIGINT` can shutdown the controller, the user's `ctrl+c` will directly exit the rs side.

Reply:
I see. It's still very weird that the rust process can get this signal, since it's already handled by the main process.

Reply:
Yes... maybe the subprocess will run independently and receive signal from the console's input.